shipyard_blueprints/email/shipyard_email/cli/send_conditional_email.py

```diff
@@ -0,0 +1,160 @@
+    InvalidInputError,
+    ConditionNotMetError,
+    InvalidCredentialsError,
+)
+from shipyard_templates import ShipyardLogger, Messaging, ExitCodeException
+
+MAX_SIZE_BYTES = 10000000
+
+logger = ShipyardLogger.get_logger()
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--send-method", dest="send_method", default="tls", required=False
+    )
+    parser.add_argument("--smtp-host", dest="smtp_host", required=True)
+    parser.add_argument("--smtp-port", dest="smtp_port", default="", required=True)
+    parser.add_argument(
+        "--sender-address", dest="sender_address", default="", required=True
+    )
+    parser.add_argument("--sender-name", dest="sender_name", default="", required=False)
+    parser.add_argument("--to", dest="to", default="", required=False)
+    parser.add_argument("--cc", dest="cc", default="", required=False)
+    parser.add_argument("--bcc", dest="bcc", default="", required=False)
+    parser.add_argument("--username", dest="username", default="", required=False)
+    parser.add_argument("--password", dest="password", default="", required=True)
+    parser.add_argument("--subject", dest="subject", default="", required=False)
+    parser.add_argument("--message", dest="message", default="", required=True)
+    parser.add_argument(
+        "--source-file-name", dest="source_file_name", default="", required=False
+    )
+    parser.add_argument(
+        "--source-folder-name", dest="source_folder_name", default="", required=False
+    )
+    parser.add_argument(
+        "--conditional-send",
+        dest="conditional_send",
+        default="always",
+        required=False,
+        choices={"file_exists", "file_dne", "always"},
+    )
+    parser.add_argument(
+        "--source-file-name-match-type",
+        dest="source_file_name_match_type",
+        default="exact_match",
+        choices={"exact_match", "regex_match"},
+        required=False,
+    )
+    parser.add_argument(
+        "--file-upload",
+        dest="file_upload",
+        default="no",
+        required=True,
+        choices={"yes", "no"},
+    )
+    parser.add_argument(
+        "--include-shipyard-footer",
+        dest="include_shipyard_footer",
+        default="TRUE",
+        choices={"TRUE", "FALSE"},
+        required=False,
+    )
+
+    args = parser.parse_args()
+    if not (args.to or args.cc or args.bcc):
+        raise InvalidInputError(
+            "Email requires at least one recipient using --to, --cc, or --bcc"
+        )
+    return args
+
+
+def main():
+    try:
+        args = get_args()
+        send_method = args.send_method.lower() or "tls"
+        sender_address = args.sender_address
+        username = args.username
+        message = args.message
+        conditional_send = args.conditional_send
+        file_upload = args.file_upload
+        if args.include_shipyard_footer:
+            include_shipyard_footer = shipyard.args.convert_to_boolean(
```

Comment:
Can we agree to not use the `convert_to_boolean` moving forward. I like the `.strip().upper() == 'TRUE'` approach

Reply:
convert_to_boolean does the `string.upper().strip()` but it is more explicit. I like the thought of being more explicit- i think changing convert_to_boolean not to raise a value error if the string is not true or false but instead return a default value would be the better approach and log a warning so the user is made aware.

```
def convert_to_boolean(string: str, default: bool = False) -> bool:
    """
    Convert a string to a boolean.

    Args:
    string (str): A string to convert to a boolean.

    Returns:
    bool: True if the string is "TRUE" (case-insensitive) else False.
    """
    logger.debug(f"Converting {string} to boolean...")

    bool_string = str(string).upper().strip()  # Handles if string is None
    if bool_string == "TRUE":
        result = True
    elif bool_string == "FALSE":
        result = False
    else:
        result = default
        logger.warning(f'"{string}" is not a valid boolean. Defaulting to {default}.')

    logger.debug(f"Converted {string} to {result} boolean.")
    return result
    ```